features/util.py

```diff
@@ -85,72 +85,87 @@ def lxc_get_property(name: str, property_name: str, image: bool = False):
 ) -> "List[str]":
     """
-    Push source PR code .tar.gz to the container.
-    Run tools/build-from-source.sh which will create the .deb
-    Copy built debs from the build container back to the source environment
-    Stop the container.
-
-    :param container_name: the name of the container to:
-         - push the PR source code;
-         - pull the built .deb package.
+    Build the package through sbuild and store the debs into
+    output_deb_dir
+
+    :param series: The target series to build the package for
     :param output_deb_dir: the target directory in which to copy deb artifacts
-    :param cloud_api: Optional pycloudlib BaseCloud api if available for the
-        machine_type
+    :param cache_source: If False, we will always rebuild the source environemt
+                         for sbuild
 
     :return: A list of file paths to debs created by the build.
-
     """
-    if os.environ.get("TRAVIS") != "true":
-        logging.info(
-            "--- Assuming non-travis build. Creating: {}".format(SOURCE_PR_TGZ)
+    logging.info("--- Creating: {}".format(SOURCE_PR_TGZ))
+    temp_dir = tempfile.gettempdir()
+    if not os.path.exists(os.path.dirname(SOURCE_PR_TGZ)):
+        os.makedirs(os.path.dirname(SOURCE_PR_TGZ))
+
+    if not os.path.exists(SOURCE_PR_TGZ) or not cache_source:
+        cwd = os.getcwd()
+        os.chdir("..")
+        subprocess.run(
+            [
+                "tar",
+                "-zcf",
+                SOURCE_PR_TGZ,
+                "--exclude-vcs",
+                "--exclude-vcs-ignores",
+                os.path.basename(cwd),
+            ]
         )
-        if not os.path.exists(os.path.dirname(SOURCE_PR_TGZ)):
-            os.makedirs(os.path.dirname(SOURCE_PR_TGZ))
-        if not os.path.exists(SOURCE_PR_TGZ) or not cache_source:
-            cwd = os.getcwd()
-            os.chdir("..")
-            subprocess.run(
-                [
-                    "tar",
-                    "-zcf",
-                    SOURCE_PR_TGZ,
-                    "--exclude-vcs",
-                    "--exclude-vcs-ignores",
-                    os.path.basename(cwd),
-                ]
-            )
-            os.chdir(cwd)
-    buildscript = "build-from-source.sh"
-    with open(buildscript, "w") as stream:
-        stream.write(BUILD_FROM_TGZ)
-
-    instance = cloud_api.get_instance(instance_id=container_name)
-    for filepath in (buildscript, SOURCE_PR_TGZ):
-        logging.info("--- Push {} -> {}:/tmp".format(filepath, instance.name))
-        instance.push_file(filepath, "/tmp/" + os.path.basename(filepath))
-    result = instance.execute(["sudo", "bash", "/tmp/" + buildscript])
-    if result.failed:
-        logging.error("--- ERROR: Building PR failed")
-        raise Exception(
-            "stdout: {}\nstderr: {}".format(result.stdout, result.stderr)
+        os.chdir(cwd)
+
+    logging.info("--- Creating: {}".format(SOURCE_PR_UNTAR_DIR))
```

Comment:
@paride I wasn't able to run sbuild using a tarball directly, that's why I am using this approach were I untar the code tarball and run sbuild passing the directory as an argument

Reply:
I'll try to figure this out.

Reply:
I can see the issue here. The normal way to produce a .deb source package composed by a .dsc file and tarball would be:
```
dpkg-source -b .
```
which automatically ignores several stuff, namely: `-I*.a -I*.la -I*.o -I*.so -I.*.sw? -I*/*~ -I,,* -I.[#~]* -I.arch-ids -I.arch-inventory -I.be -I.bzr -I.bzr.backup -I.bzr.tags -I.bzrignore -I.cvsignore -I.deps -I.git -I.gitattributes -I.gitignore -I.gitmodules -I.gitreview -I.hg -I.hgignore -I.hgsigs -I.hgtags -I.mailmap -I.mtn-ignore -I.shelf -I.svn -ICVS -IDEADJOE -IRCS -I_MTN -I_darcs -I{arch}`.
This list doesn't cover `.tox` and there is no "smart" option to parse `.gitignore` like `tar --exclude-vcs-ignores`. There is also no way to pass raw options to tar.

We could specify what to ignore manually via `-I`, but it will get out of sync at some point. I have another suggestion:
```
git archive --prefix=ua-clean-tree/ HEAD | tar -x -C ..
```
this should export whatever is committed into git HEAD to `../ua-clean-tree/`, without any `.git` or uncommitted/untracked filed. We can then sbuild that directory.

Reply:
One downside to this approach is that it will not capture any untracked changes, if i understand correctly. We would like to easily test changes that are in-progress using the integration tests. With this approach, we can do that, we just need to create a commit for every change before running the tests with `BUILD_PR=1`.

This isn't a deal-breaker, but if we go this route, then we should document this in the readme.

Reply:
@paride I think the problem is not generating the tarball per se, but using it directly on sbuild, like:
`sbuild uaclient.tar.gz`

It seems that sbuild does not accept the tarball directly. When I used this, it tried to download a package source with that name instead of using the tarball directly.

But I can be missing something here

Reply:
Yep we need the full source package (tarball + .dsc), that's what `dpkg-source -b` would do, but it isn't as smart as `tar --exclude-vcs-ignores`. Per standup discussion, I'm fine with the tarball approach at this point.

Maybe we could do `tar -c | tar -x -C` without ever writing the tarball to disk, for a *slight* performance gain. Not sure if worth it, up to you. Thanks for implementing this!